libs/stats/odc/stats/_cli_save_tasks.py

```diff
@@ -151,3 +162,38 @@ def gqa_predicate(ds):
     if not ok:
         # exit with error code, failure message was already printed
         sys.exit(3)
+
+
+def _parse_products(dc, products, temporal_range):
+    if len(products) == 1:
+        product = products[0]
+        dss = None
+        n_dss = None
+    elif len(products) == 2:
+        
+        query = {"product": products}
+        query.update(
+                temporal_range.dc_query(pad=0.6)
+            ) 
+        dss = ordered_dss(
+            dc, key=lambda ds: (ds.center_time, ds.metadata.region_code), **query
+        )
+        paired_dss = groupby(dss, key=lambda ds: (ds.center_time, ds.metadata.region_code))
+        paired_dss = [list(t[1]) for t in paired_dss]
```

Comment:
this will run out of RAM on large enough queries, it really needs to be a streaming operation